Library/Homebrew/service.rb

```diff
@@ -100,15 +100,31 @@ def error_log_path(path = nil)
       end
     end
 
-    sig { params(value: T.nilable(T::Boolean)).returns(T.nilable(T::Boolean)) }
+    sig { params(value: T.nilable(T.any(T::Boolean, T::Hash))).returns(T.nilable(T::Hash)) }
     def keep_alive(value = nil)
       case T.unsafe(value)
       when nil
         @keep_alive
       when true, false
+        @keep_alive = { always: value }
+      when Hash
         @keep_alive = value
       else
-        raise TypeError, "Service#keep_alive expects a Boolean"
+        raise TypeError, "Service#keep_alive expects a Boolean or Hash"
+      end
+    end
+
+    sig { params(value: T.nilable(String)).returns(T.nilable(T::Hash[Symbol, String])) }
+    def sockets(value = nil)
+      case T.unsafe(value)
+      when nil
+        @sockets
+      when String
+        type,value = value.split('://')
+        host,port = value.split(':')
+        @sockets = {host: host, port: port, type: type, ipv: 4}
```

Comment:
Feels like more input validation is needed here to ensure the format is correct. I'd recommend using a regex with match groups.